From 8531b38fb5a96a9b7736b6d031e32bcbd877d91a Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Mon, 19 Dec 2016 23:02:37 +0100
Subject: [PATCH 10/13] usb-storage: add an option to disable support for
 unusual devices

Most of them are pretty exotic or old. Disabling this saves 4kB in
the xz-compressed image.
---
 drivers/usb/storage/Kconfig        | 9 +++++++++
 drivers/usb/storage/unusual_devs.h | 2 ++
 2 files changed, 11 insertions(+)

diff --git a/drivers/usb/storage/Kconfig b/drivers/usb/storage/Kconfig
index 1dd0604..e53a54c 100644
--- a/drivers/usb/storage/Kconfig
+++ b/drivers/usb/storage/Kconfig
@@ -25,6 +25,15 @@ config USB_STORAGE
 	  To compile this driver as a module, choose M here: the
 	  module will be called usb-storage.
 
+config USB_STORAGE_UNUSUAL_DEVS
+	bool "USB Mass Storage unusual device support"
+	depends on USB_STORAGE
+	help
+	  Say Y here in order to properly support the widest range of devices,
+	  or N to save about 4kB of image size. Some old or exotic devices need
+	  this option and some pretty common one as well. It's likely that
+	  without it, some USB flash sticks and some hard drivers may not work.
+
 config USB_STORAGE_DEBUG
 	bool "USB Mass Storage verbose debug"
 	depends on USB_STORAGE
diff --git a/drivers/usb/storage/unusual_devs.h b/drivers/usb/storage/unusual_devs.h
index da380a9..184dc6b 100644
--- a/drivers/usb/storage/unusual_devs.h
+++ b/drivers/usb/storage/unusual_devs.h
@@ -61,6 +61,7 @@
  * maintained at http://www.draisberghof.de/usb_modeswitch/
  */
 
+#ifdef CONFIG_USB_STORAGE_UNUSUAL_DEVS
 #if !defined(CONFIG_USB_STORAGE_SDDR09) && \
 		!defined(CONFIG_USB_STORAGE_SDDR09_MODULE)
 #define NO_SDDR09
@@ -2156,6 +2157,7 @@ UNUSUAL_DEV( 0xed10, 0x7636, 0x0001, 0x0001,
 		"TGE",
 		"Digital MP3 Audio Player",
 		USB_SC_DEVICE, USB_PR_DEVICE, NULL, US_FL_NOT_LOCKABLE ),
+#endif
 
 /* Control/Bulk transport for all SubClass values */
 USUAL_DEV(USB_SC_RBC, USB_PR_CB),
-- 
1.7.12.1

